src/input_common/sdl/sdl_impl.h

```diff
@@ -9,6 +9,17 @@
 #include <mutex>
 #include <thread>
 #include <unordered_map>
+
+// Ignore -Wimplicit-fallthrough due to https://github.com/libsdl-org/SDL/issues/4307
+#ifdef __GNUC__
+#pragma clang diagnostic push
```

Comment:
```suggestion
#pragma GCC diagnostic push
```

Ditto for the other pragmas. So that this works for both GCC and clang.

This include can also probably be kept in the cpp file, by just making the class member functions and types that use it internal to the cpp file. 

e.g. Doing a quick glance over the added private functions, I couldn't see any place where the functions depend on private class state, so they can probably just be moved into an anonymous namespace. If there is class state, then the class can pass it to those functions at their respective call sites.